org.eclipse.ui.genericeditor/src/org/eclipse/ui/internal/genericeditor/GenericContentTypeRelatedExtension.java

```diff
@@ -54,9 +55,13 @@ public GenericContentTypeRelatedExtension(IConfigurationElement element) throws
 	}
 
 	@SuppressWarnings("unchecked")
-	public T createDelegate() {
+	public T createDelegate(ITextEditor editor) {
 		try {
-			return (T) extension.createExecutableExtension(CLASS_ATTRIBUTE);
+			T delegateInstance = (T) extension.createExecutableExtension(CLASS_ATTRIBUTE);
+			if (delegateInstance instanceof ITextEditorAware) {
```

Comment:
Since the API ITextEditorAware was added in ui.workbench.texteditor, I'd expect it to be respected there, too. Right now, it looks as if something was added in ui.workbench just for the sake of ui.genericeditor. This has a bad smell to me.

Reply:
I agree with you @szarnekow I think more and more that this API should be hosted in generic editor. 

I think too ItextViewerLyfecicle https://github.com/eclipse-platform/eclipse.platform.text/pull/107 should be hosted in generic editor

@mickaelistria are you OK with that.